api/ipfsproxy/config.go

```diff
@@ -116,26 +128,49 @@ func (cfg *Config) LoadJSON(raw []byte) error {
 		return err
 	}
 
-	cfg.Default()
+	if jcfg.ListenMultiaddress == "" {
```

Comment:
It would be good to put the backwards compat `if` statements in a function so they don't clutter up the function and it makes it clear what they are.